### PR TITLE
Integrate fedora adapter

### LINKED ISSUE #12: Integrate ember-fedora-adapter
Setup Mirage to mock JSON-LD data. This will also help enable automated tests

### PR DESCRIPTION
**Pull request by jabrah**
Closes #12 

GET requests should work, other requests may not work quite right. Getting POSTs, PATCH, etc working correctly can be done as needed.

* Initial integration of `ember-fedora-adapter`
* Change Mirage setup to emulate the setup from PASS
* Remove abandoned Mirage stuff
* Adds some groundwork for Docker development environment, but the Docker env currently doesn't give anything useful.

### CODE REVIEW

File: .docker/secrets/idp/idp-encryption.key [hunk 1]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (jaredgalanis) [inline .docker/secrets/idp/idp-encryption.key]: Should this be checked in?

File: app/models/file.js [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
-export default Model.extend({
-  name: DS.attr('string'),
-  externalId: DS.attr('string'),
-  description: DS.attr('string'),
-
-  /** processing, changesRequired, accepted, published */
-  status: DS.attr('string')
-});
+export default class File extends Model {
+  @attr('string') name;
```
Comment (jaredgalanis) [inline app/models/file.js]: Nice :) I had not even thought to upgrade the Model files to class and decorator syntax

File: .docker/secrets/idp/idp-encryption.key (hunk 1 above)
Comment (jabrah) [reply .docker/secrets/idp/idp-encryption.key]: These `secrets` files are copied from `jhuda-docker` repo and is also the same as is used in the local PASS dev environment. These keys are used only for local development environments

**Comment by jabrah** [comment]
@jaredgalanis I'd like a brief re-review, focusing on the changes made to `metadata-form-tests`. The previous interaction with Mirage creating objects, then retrieving them from the store was not working with the way `ember-fedora-adapter` wanted to do things. Just want to make sure writing the tests this way still meshes with the intent of the tests

**Comment by jaredgalanis** [comment]
> @jaredgalanis I'd like a brief re-review, focusing on the changes made to `metadata-form-tests`. The previous interaction with Mirage creating objects, then retrieving them from the store was not working with the way `ember-fedora-adapter` wanted to do things. Just want to make sure writing the tests this way still meshes with the intent of the tests

@jabrah Looks good to me. I don't see an issue with not using mirage to populate the test server db. The way I see it, is that was a brief stand-in for what will ultimately be the serializing that needs to occur into and out of the `submission.metadata` jsonb field. I think once we have that working it will likely mean there aren't back queries during this step to build up those models.

**Pull request merged**